### PR TITLE
Move asserts to contracts and give diagnostics

### PR DESCRIPTION
**Pull request by jmh530**
I wanted to improve diagnostics by moving the asserts into in contracts and adding in language for them. 

A next step could be adding additional asserts for functions that are light on them.

### CODE REVIEW

**Comment by 9il** [comment]
Could you please convert assert msgs to a single line. Code line can contain as many symbols as needed. Something like:
```
    assert(cond,  "small msg");
    assert(cond,  "biiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiig msg");
or
    assert(cond, "biiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiii" ~
    "iiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiig msg");


```

File: source/mir/lapack.d [hunk 1]
```diff
@@ -18,18 +18,16 @@ static import lapack;
...
-    squareM = "The matrix must be square"
-};
-
 @trusted pure nothrow @nogc:
 
 /// `getri` work space query.
 size_t getri_wq(T)(Slice!(T*, 2, Canonical) a)
+in
+{
+    assert(a.length!0 == a.length!1, "getri: a must be a square matrix.");
```
Comment (thewilsonator) [inline source/mir/lapack.d]:
```
"getri: `a` must be a square matrix."
```

differentiate the parameter from the indefinite article. Probably could do with emphasis on the parameters in the other assert messages as well, but this one is more important.

Comment (jmh530) [reply source/mir/lapack.d]: Is it ok to do it as 'a', or does it have to be `a`?

Comment (thewilsonator) [reply source/mir/lapack.d]: Is for asserts, so it doesn't really matter.

**Comment by 9il** [comment]
travis fails

**Pull request merged**